openmdao/core/component.py

```diff
@@ -492,6 +498,10 @@ def add_input(self, name, val=1.0, shape=None, src_indices=None, flat_src_indice
         metadata['desc'] = desc
         metadata['distributed'] = self.options['distributed']
 
+        if isinstance(tags, str):
+            tags = [tags, ]
+        metadata['tags'] = tags
```

Comment:
We use tags later as a set, so we should just store them internally as a set to begin with.  Also, turning tags into a set removes the reference to the tags list that gets passed in.  That's a good thing since that list could get modified later outside of this component.